src/modules/direction/components/DirectionView.tsx

```diff
@@ -46,21 +46,30 @@ const DirectionView: React.FC<IDirectionViewProps> = () => {
     dispatch(
       fetchExperts(currentDirection.name, currentDirection.id as number),
     );
-  }, []);
-
-  useEffect(() => {
     dispatch(fetchCourses(currentDirection.name));
-  }, []);
 
-  const courseCards = useSelector(
-    (state: RootStateType) => state.directions[currentDirection.name]?.courses,
-  );
+    return () => {
+      dispatchFilters([], currentDirection.name);
+      dispatch(
+        setPostFilters({
+          key: FilterTypeEnum.TAGS,
+          filters: {
+            value: [],
+          },
+          directionName: currentDirection.name,
+        }),
+      );
+    };
```

Comment:
resetting the filters when changing the view, right?

do you think it's necessary?

Reply:
yes. that seemed reasonable, works like that on most webpages. 
checkboxes and tags inputs were unset after changing views, meaning their displayed state didn't match the one saved in redux, so we needed to either reset filters when leaving page or fix filters not displaying correct values when we visit the page again.